### ISSUE TITLE
Generic geometry type

### ISSUE THREAD

**Issue by MarAlder** [DLR-boeh-da/CPACS#844]
Should we introduce a `genericGeometryType` that can essentially be composed of fuselage and wing-like lofts? This would be particularly suitable for the predefinable component library (`systemElements`, `cabinElements`, in future: `payloadElements`).

![grafik](https://github.com/user-attachments/assets/0aa4923b-5cdb-42af-93d3-4da37ae18513)

Note: not sure about a proper naming of such bodies, though 🤔

**Comment by sdeinert** [comment]
Sounds quite useful!
I am just wondering whether we would need wing-like lofts... Wouldn't it be sufficient to have fuselage like bodies that can be rotated freely. Wings are only a limitation of a fuselage loft.
Combined with other standard geometries as for the genericSystems (sphere, cube, ...) this would cover a great range of possible geometries.
Then also the distinction between transverse and longitudinal bodies would not really be required.

One risk that I see is that these generic geometries could then be used also for components that have an actual definition in CPACS. So we might need to think of restricting the usage somehow to avoid fuselages, ducts, systems, etc. being defined through genericGeometries.

**Comment by MarAlder** [comment]
> I am just wondering whether we would need wing-like lofts... Wouldn't it be sufficient to have fuselage like bodies that can be rotated freely. Wings are only a limitation of a fuselage loft.

Good point, I didn't know that. I'll have a look at the TiGL implementation and make a prototype first anyway.

> One risk that I see is that these generic geometries could then be used also for components that have an actual definition in CPACS. So we might need to think of restricting the usage somehow to avoid fuselages, ducts, systems, etc. being defined through genericGeometries.

My idea was to use the type for explicit elements, see #845. At this level, there is of course the discussion of whether we have `genericComponents` or such which might be misused.

**Comment by MarAlder** [comment]
Continued in #848
